### PR TITLE
Search bar

### LINKED ISSUE #86: search bar on desktop?
what do we want to do with the empty space? I though it would initiate when user scroll below the search form below the VIKING GARAGE header. How do you see that?

### PR DESCRIPTION
**Pull request by jonzee**
#86
- added search bar on desktop with `Location` filter logic
- restructured project (added wrapper with data on top level)
- rewrite some parts of `Search` and `Filters`

### CODE REVIEW

**Comment by michalmikolajczyk** [comment]
we can resolve the conflicts together using Kaleidoscope

File: src/ts/typings/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export interface IDate {
```
Comment (jonzee) [inline src/ts/typings/index.ts]: `I` > `Interface`, `T` > `Type` etc., to keep namespace different than components (`Date`, `Location` etc.)

File: src/ts/Container/index.tsx [hunk 2]
```diff
@@ -4,22 +4,104 @@ import {
...
-    const userAgent = typeof navigator !== 'undefined'
-      ? navigator.userAgent
-      : _.get(this.context, 'data.userAgent', 'all');
+    const props = {
+      ...this.state,
+      loadMore: this.loadMore,
+      filtersFuncs: this.filters,
+      filtersValue: this.state.filters,
+    };
+    const childrenWithProps = React.cloneElement(this.props.children, props);
```
Comment (michalmikolajczyk) [inline src/ts/Container/index.tsx]: is this recommended?

File: src/ts/Offers/OffersList.tsx [hunk 3]
```diff
@@ -6,15 +6,15 @@ import Offer from './Offer';
...
     date,
+    loading,
     location,
   } = props;
 
   const items = list.map((item, key) => <Offer data={item} key={key} date={date} location={location}/>);
 
   return (
-    <div className={`offers ${load ? 'load' : ''}`}>
+    <div className={`offers ${loading ? 'loading' : ''}`}>
```
Comment (michalmikolajczyk) [inline src/ts/Offers/OffersList.tsx]: use some lodash one-liner

File: src/ts/Search/FullText.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+export default class FullText extends React.Component<FullTextProps, any> {
+  state = {
+    data: [],
+    open: false,
+    value: '',
+  };
+  dataConfig = { text: 'description', value: 'place_id' };
+  statusOk = typeof google !== 'undefined' ? google.maps.places.PlacesServiceStatus.OK : null;
+  placesService = typeof google !== 'undefined' ? new google.maps.places.PlacesService(document.createElement('div')) : null;
```
Comment (michalmikolajczyk) [inline src/ts/Search/FullText.tsx]: length of line

File: src/ts/Search/FullText.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  selectService = typeof google !== 'undefined' ? new google.maps.places.AutocompleteService() : null;
+
+  toggle = () => this.setState({ open: !this.state.open });
+
+  onNewRequest = (details, index) => {
```
Comment (michalmikolajczyk) [inline src/ts/Search/FullText.tsx]: please return a value

File: src/ts/Search/Type.tsx [hunk 6]
```diff
@@ -39,14 +47,17 @@ export default class Select extends React.Component<any, any> {
     }
   })
 
-  isGroup(str: String): boolean {
+  isGroup(str: string): boolean {
```
Comment (jonzee) [inline src/ts/Search/Type.tsx]: nope, it's a proper way of typings https://stackoverflow.com/a/14727461/6041704

File: src/ts/Search/Type.tsx [hunk 7]
```diff
@@ -39,14 +47,17 @@ export default class Select extends React.Component<any, any> {
...
     return str.indexOf('#') === -1;
   }
 
   save = (values) => {
```
Comment (jonzee) [inline src/ts/Search/Type.tsx]: it's necessary with airbnb styleguide

File: src/ts/Offers/OffersList.tsx (hunk 3 above)
Comment (jonzee) [inline src/ts/Offers/OffersList.tsx]: couldn't find any reasonable way to change this

Comment (michalmikolajczyk) [inline src/ts/Offers/OffersList.tsx]: my bad, missed the quotes

**Pull request merged**